src/components/radio/daikin-radio.ts

```diff
@@ -0,0 +1,226 @@
+import ctl from "@netlify/classnames-template-literals";
+import { css, html, LitElement, unsafeCSS } from "lit";
+import { customElement, property } from "lit/decorators.js";
+
+import { cva, type VariantProps } from "class-variance-authority";
+import tailwindStyles from "../../tailwind.css";
+import type { OmitNull } from "../../typeUtils";
+
+const hideRadioCN = ctl(`
+  opacity-0
+  absolute
+  peer
+`);
+
+const labelCN = cva(
+  ["leading-8", "not-italic", "font-normal", "align-middle"],
+  {
+    variants: {
+      size: {
+        small: ["text-sm"],
+        large: ["text-base"],
+      },
+    },
+    defaultVariants: {
+      size: "small",
+    },
+  }
+);
+
+const radioCN = cva(
+  [
+    "relative",
+    "after:absolute",
+    "after:!w-full",
+    "after:!h-full",
+    "after:i-daikin-radio-unchecked",
+    "after:text-[#8C8C8C]",
+    "peer-checked:after:i-daikin-radio-checked",
+    "peer-checked:after:text-daikinBlue-500",
+
+    "peer-hover:after:i-daikin-radio-checked",
+    "peer-hover:after:text-daikinBlue-300",
+
+    "peer-active:after:i-daikin-radio-checked",
+    "peer-active:after:text-daikinBlue-500",
+
+    "peer-focus-visible:after:i-daikin-radio-unchecked",
+    "peer-focus-visible:peer-checked:after:i-daikin-radio-checked",
+    "peer-focus-visible:after:text-daikinBlue-700",
+
+    "peer-disabled:after:i-daikin-radio-unchecked",
+    "peer-disabled:peer-checked:after:i-daikin-radio-checked",
+    "peer-disabled:after:text-daikinNeutral-200",
+  ],
+  {
+    variants: {
+      size: {
+        small: ["w-[14px]", "h-[14px]"],
+        large: ["w-4", "h-4"],
+      },
+    },
+    defaultVariants: {
+      size: "small",
+    },
+  }
+);
+
+type LabelProps = OmitNull<VariantProps<typeof labelCN>>;
+type RadioProps = OmitNull<VariantProps<typeof radioCN>>;
+type ComponentSizeProps = LabelProps["size"] & RadioProps["size"];
+
+export interface DaikinRadioProps {
+  label: string;
+  size: ComponentSizeProps;
+  disabled: boolean;
+  labelPosition: "left" | "right";
+  readonly: boolean;
+  checked: boolean;
+  name: string;
+  value: string;
+  error: boolean;
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+@customElement("daikin-radio")
+class DaikinRadio extends LitElement implements DaikinRadioProps {
+  static styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  private _handleClick(event: MouseEvent) {
+    if (this.readonly) {
+      event.preventDefault();
+    }
+  }
+
+  static formAssociated = true;
+
+  // define internals to let radio can be used in form
+  @property({ type: Object })
+  internals;
+
+  constructor() {
+    super();
+    this.internals = this.attachInternals();
+  }
+
+  private _updateFormValue() {
+    if (this.checked) {
+      this.internals.setFormValue(this.value);
+    } else {
+      this.internals.setFormValue(null);
+    }
```

Comment:
Might be better to make it common whenever possible.

```suggestion
    this.internals.setFormValue(this.checked ? this.value : null);
```